community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -775,23 +765,15 @@ sealed class TransactionBoundQueryContext(val transactionalContext: Transactiona
       }
     }
 
-    override def indexGet(name: String, key: String, value: Any): Iterator[RelationshipValue] =
-      JavaConversionSupport.mapToScalaENFXSafe(
-        transactionalContext.statement.readOperations().relationshipExplicitIndexGet(name, key, value, -1, -1))(getByIdIfExists)
-
-    override def indexQuery(name: String, query: Any): Iterator[RelationshipValue] =
-      JavaConversionSupport.mapToScalaENFXSafe(
-        transactionalContext.statement.readOperations().relationshipExplicitIndexQuery(name, query, -1, -1))(getByIdIfExists)
-
     override def isDeletedInThisTx(id: Long): Boolean =
       transactionalContext.stateView.hasTxStateWithChanges && transactionalContext.stateView.txState()
         .relationshipIsDeletedInThisTx(id)
 
     override def acquireExclusiveLock(obj: Long): Unit =
-      transactionalContext.statement.readOperations().acquireExclusive(ResourceTypes.RELATIONSHIP, obj)
+      transactionalContext.kernelTransaction.locks().acquireExclusiveRelationshipLock(obj)
 
     override def releaseExclusiveLock(obj: Long): Unit =
-      transactionalContext.statement.readOperations().acquireExclusive(ResourceTypes.RELATIONSHIP, obj)
+      transactionalContext.kernelTransaction.locks().acquireExclusiveRelationshipLock(obj)
```

Comment:
Don't know this has been working (never used?), but this should definitely be a `releaseExclusiveRelationshipLock`